ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -347,10 +348,46 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	case rayv1.JobDeploymentStatusComplete, rayv1.JobDeploymentStatusFailed:
 		// If this RayJob uses an existing RayCluster (i.e., ClusterSelector is set), we should not delete the RayCluster.
 		logger.Info(string(rayJobInstance.Status.JobDeploymentStatus), "RayJob", rayJobInstance.Name, "ShutdownAfterJobFinishes", rayJobInstance.Spec.ShutdownAfterJobFinishes, "ClusterSelector", rayJobInstance.Spec.ClusterSelector)
-		if rayJobInstance.Spec.ShutdownAfterJobFinishes && len(rayJobInstance.Spec.ClusterSelector) == 0 {
-			ttlSeconds := rayJobInstance.Spec.TTLSecondsAfterFinished
-			nowTime := time.Now()
-			shutdownTime := rayJobInstance.Status.EndTime.Add(time.Duration(ttlSeconds) * time.Second)
+		ttlSeconds := rayJobInstance.Spec.TTLSecondsAfterFinished
+		nowTime := time.Now()
+		shutdownTime := rayJobInstance.Status.EndTime.Add(time.Duration(ttlSeconds) * time.Second)
+
```

Comment:
maybe move the check before `if features.Enabled(features.RayJobDeletionPolicy) ...` to simplify the code?

```go
	if shutdownTime.After(nowTime) {
		delta := int32(time.Until(shutdownTime.Add(2 * time.Second)).Seconds())
		logger.Info("shutdownTime not reached; requeue this RayJob.", "requeue seconds", delta)
		return ctrl.Result{RequeueAfter: time.Duration(delta) * time.Second}, nil
	}
```

Reply:
This could add a requeue for cases where deletion is not required